assets/charts/control-plane/kube-apiserver/templates/kube-apiserver-ds.yaml

```diff
@@ -0,0 +1,29 @@
+{{- if gt (int .Values.apiserver.replicas) 1 }}
+apiVersion: apps/v1
+kind: DaemonSet
```

Comment:
There is one disadvantage of running those components as DaemonSet instead of Deployment, which I recently hit myself.

If one runs 3 node controller pool, but each node is resource constrained (e.g. 2 vCPUs, 4GB of RAM), then one might want to run only 2 replicas of each component to save a bit of resources, as for example each kube-apiserver is very RAM-costly.

I think it should be our conscious decision, that we won't be able to support such scenario.

Reply:
I would rather run less number of other components than APIServer. If I were to run my prod cluster then I should make sure that control plane is not fighting for resources. On the workers I will try to bin pack my work loads, not on the control plane.

Reply:
Leaving this open so that we know this was decided consciously.